tracex_project/extraction/forms.py

```diff
@@ -121,6 +145,69 @@ def get_patient_journey_choices():
 class FilterForm(BaseEventForm):
     """Form for selecting filter for extraction result"""
 
+    def clean(self):
+        """Validate form data."""
+        cleaned_data = super().clean()
+        modules = cleaned_data.get("modules_optional") + cleaned_data.get(
+            "modules_required"
+        )
+        activity_key = cleaned_data.get("activity_key")
+        key_to_module = {
+            "event_type": "event_type_classification",
+            "activity": "activity_labeling",
+            "attribute_location": "location_extraction",
+        }
+        if key_to_module[activity_key] not in modules:
+            error_module = [
+                module[1]
```

Comment:
for readability I would put module[1] and module[0] in declarative variables so we know what we are working with